crates/sui-replay/src/replay.rs

```diff
@@ -2128,3 +2176,38 @@ pub fn get_executor(
     sui_execution::executor(&protocol_config, silent, enable_profiler)
         .expect("Creating an executor should not fail here")
 }
+
+fn parse_effect_error_for_denied_coins(status: &SuiExecutionStatus) -> Option<String> {
+    let SuiExecutionStatus::Failure { error } = status else {
+        return None;
+    };
+    parse_denied_error_string(error)
+}
+
+fn parse_denied_error_string(error: &str) -> Option<String> {
+    let regulated_regex = regex::Regex::new(
```

Comment:
This makes me sad, but at least currently there's no other way of doing this other than to regex the string for it since execution errors are just a `String` in the transaction effects response :'(

Reply:
We can fix this more readily in GraphQL (offer a structured error response).